sdk/azidentity/internal/customtokenproxy/policy.go

```diff
@@ -0,0 +1,231 @@
+
+	resp, err := tr.RoundTrip(rawReq)
+	if err == nil && resp == nil {
+		// this policy is effectively a transport, so it must handle
+		// this rare case. Returning an error makes the retry policy
+		// try the request again
+		err = errors.New("received nil response")
+	}
+	return resp, err
+}
+
+// getTokenTransporter provides the token transport to use for the request.
+//
+// There are a few scenarios need to be handled:
+//  1. no CA overrides, use default transport
+//  2. CA data override provided, use a transport with custom CA pool.
+//     This transport is fixed after set.
+//  3. CA file override is provided, use a transport with custom CA pool.
+//     This transport needs to be recreated if the CA file content changes.
+func (p *customTokenProxyPolicy) getTokenTransporter() (*http.Transport, error) {
+	if len(p.caData) == 0 && p.caFile == "" {
+		// no custom CA overrides
+		if p.transport == nil {
+			p.transport = createTransport(p.sniName, nil)
+		}
+		return p.transport, nil
+	}
+
+	if p.caFile == "" {
+		// host provided CA bytes in AZURE_KUBERNETES_CA_DATA and can't change
+		// them now, so we need to create a client only if we haven't done so yet
+		if p.transport != nil {
+			return p.transport, nil
+		}
+
+		caPool := x509.NewCertPool()
+		if !caPool.AppendCertsFromPEM([]byte(p.caData)) {
+			return nil, fmt.Errorf("parse CA data: no valid certificates found")
+		}
+
+		p.transport = createTransport(p.sniName, caPool)
+		return p.transport, nil
+	}
+
+	// host provided the CA bytes in a file whose contents it can change,
+	// so we must read that file and maybe create a new client
+	b, err := os.ReadFile(p.caFile)
+	if err != nil {
+		return nil, fmt.Errorf("read CA file %q: %s", p.caFile, err)
+	}
+	if len(b) == 0 {
+		// this can happen during the middle of CA rotation on the host.
+		if p.transport == nil {
+			// if the transport was never created, error out here to force retrying the call later
+			return nil, fmt.Errorf("CA file %q is empty", p.caFile)
+		}
+		// if the transport was already created, just keep using it
+		return p.transport, nil
+	}
+	if !bytes.Equal(b, p.caData) {
+		// CA has changed, rebuild the transport with new CA pool
+		// invariant: p.transport is nil when p.caData is nil (initial call)
+		caPool := x509.NewCertPool()
+		if !caPool.AppendCertsFromPEM([]byte(b)) {
+			return nil, fmt.Errorf("parse CA file %q: no valid certificates found", p.caFile)
+		}
+		p.transport = createTransport(p.sniName, caPool)
+		p.caData = b
+	}
+
+	return p.transport, nil
+}
+
+// rewriteProxyRequestURL updates the request URL to target the specified URL.
+// Target is the token proxy URL in custom token endpoint mode.
+func rewriteProxyRequestURL(req *http.Request, proxyURL *url.URL) {
+	source := proxyURL
+	if source.EscapedPath() == "" {
+		// if the proxy URL has no path, set it to "/" to avoid stripping the path from req.URL
+		*source = *proxyURL
+		source.Path = "/"
+	}
```

Comment:
This looks like it would simplify to
```suggestion
	if proxyURL.EscapedPath() == "" {
		proxyURL.Path = "/"
	}
```
where `proxyURL` is the parsed value of AZURE_KUBERNETES_TOKEN_PROXY i.e. `customTokenProxyPolicy.tokenProxy`. So, can we just do this once in `parseAndValidateCustomTokenProxy`?

Reply:
good point, updated to prepare the raw path only once